pages/_app.tsx

```diff
@@ -13,8 +19,44 @@ export type AppProps = NextAppProps & {
   err?: Error;
 };
 
+function navigatorLanguage() {
+  return process.browser ? navigator.language.split("-")[0] : "en";
+}
+function useViewerLocale() {
+  const router = useRouter();
+
+  const meQuery = useMeQuery();
+  const viewerPreferredLanguage = meQuery.data?.locale ?? navigatorLanguage();
+
+  // switch locale to viewer language on first mount
+  useEffect(() => {
+    const currentLocale = router.locale;
+    // console.debug("locale check", { currentLocale, viewerPreferredLanguage });
+
+    if (currentLocale === viewerPreferredLanguage) {
+      return;
+    }
+    router.replace(
+      {
+        pathname: router.pathname,
+        query: router.query,
+      },
+      undefined,
+      {
+        locale: viewerPreferredLanguage,
+      }
+    );
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [viewerPreferredLanguage]);
```

Comment:
Hook that switches locale whenever we detect a difference between current locale and what user prefers using 

1. user settings
2. browser's settings